analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/SymbolReferenceAnalyzerTest.cs

```diff
@@ -212,6 +215,61 @@ public void Verify_UnchangedFiles(string unchangedFileName, bool expectedProtobu
                         VerifyReferences(orderedSymbols[1].Reference, 9, 44, 41);           // LocalMethod
                     });
 
+        [DataTestMethod]
+        [DataRow(ProjectType.Product)]
+        [DataRow(ProjectType.Test)]
+        public void Verify_PrimaryConstructor_PreciseLocation_CSharp12(ProjectType projectType) =>
+            Verify("PrimaryConstructor.cs", projectType, references =>
+            {
+                references.Select(x => x.Declaration.StartLine).Should().BeEquivalentTo(new[] { 1, 3, 6, 6, 8, 8, 10, 11, 12, 14, 17, 17, 19, 20, 21, 21, 23, 23, 25 });
+
+                var primaryCtorParameterDeclaration = references.Single(x => x.Declaration.StartLine == 8 && x.Declaration.StartOffset == 19); // b1, primary ctor
+                primaryCtorParameterDeclaration.Declaration.Should().BeEquivalentTo(new TextRange { StartLine = 8, EndLine = 8, StartOffset = 19, EndOffset = 21 });
+                primaryCtorParameterDeclaration.Reference.Should().BeEquivalentTo(
+                    new RepeatedField<TextRange>
+                    {
+                        new TextRange { StartLine = 10, EndLine = 10, StartOffset = 24, EndOffset = 26 }, // Field
+                        new TextRange { StartLine = 11, EndLine = 11, StartOffset = 41, EndOffset = 43 }, // Property
+                        new TextRange { StartLine = 12, EndLine = 12, StartOffset = 21, EndOffset = 23 }  // b1
+                    });
+
+                var ctorDeclaration = references.Single(x => x.Declaration.StartLine == 8 && x.Declaration.StartOffset == 6); // SubClass
+                ctorDeclaration.Reference.Should().BeEmpty(); // FN, not reporting constructor 'SubClass' and 'this' (line 14)
```

Comment:
line 14 contains `int Method() => b1;`: I don't see the reference to `this` in `Subclass`. In general, I am not sure I understand what symbol reference is being asserted here: if I select `Subclass` in Visual Studio (line 8 offset 6), I don't have anything selected:
![image](https://github.com/SonarSource/sonar-dotnet/assets/2676742/a68df3ce-4603-4e91-8ec6-781e20664c97)

I would rather assert on `StartOffset = 19` (`b1`), that should have references on lines 10, 11 and 12:
![image](https://github.com/SonarSource/sonar-dotnet/assets/2676742/f1b93d4d-f6e1-4649-b7b4-fa8e6c48fdcc)

Reply:
On the other hand, if I am misunderstanding the test case, and the false negative does exist, an issue should be created in the backlog.

Reply:
Sorry for the confusion: the ctorDeclaration test case was meant for SubClass initially, there was a ctor with this similar to the one in B class. Then I moved some tests adding a second class `B`, I should have moved the test there but I forgot. Thanks for noticing, I'll change it right away and it will be clearer, let me know then if you have any other questions

Reply:
That makes sense, thanks for clarifying!